src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -372,6 +376,8 @@ private AIState lookForNeededItems()
 
     /**
      * Updates the itemsCurrentlyNeeded with current values.
+     * 
+     * This function seems bugged, as it gives itemsCurrentlyNeeded = isEmpty 
```

Comment:
Can you elaborate why `itemsCurrentlyNeeded = new ArrayList<>(job.getItemsNeeded());` is "always empty"? How did you get that assumption and how should we fix that?